### DISCUSSION TITLE
Is the high latency setting a bad influence to SRT streaming?

### DISCUSSION THREAD

**Discussion by exp256** [Haivision/srt#2949] [Q&A] [answered]
I'd like to ask a question to isolate the source of my problem.
I'm trying SRT streaming by using gstreamer.(input--->ENC(H.264 mpegts)--->SRT stream--->DEC--->output)
When I increase the latency(property of srtsink and srtsrc) in seconds (e.g. latency=5000),
DEC output video becomes interrupted frequently.

Version
gstreamer:v1.20.5
srt:v1.5.1

**Comment by ethouris** [comment, marked as answer]
Theoretically there's nothing wrong with it, but note that the default value of the sender and receiver buffers are intended for low latency. Higher latency may be held up, but in this case you need to set up a bigger receiver buffer capacity (SRTO_RCVBUF). You can find some information here how to calculate the required capacity: https://github.com/Haivision/srt/blob/master/docs/API/API-socket-options.md#srto_fc

Too small buffer capacity will simply result in dropping packets that can't be put there and dropping them, if they can't be recovered in time, as well as breaking the connection if that has led to irrecoverable sequence discrepancy.

**Reply by exp256** [reply]
Thank you for your answer.  I solved the problem by setting bigger SRTO_RCVBUF and SRTO_FC. 
Also, increasing the buffers of gstreamer elements were neeed, too.
